dartagnan/src/main/java/com/dat3m/dartagnan/asserts/AssertBasic.java

```diff
@@ -35,4 +38,16 @@ private String valueToString(LastValueInterface value){
         }
         return value.toString();
     }
+
+	@Override
+	public List<Register> getRegs() {
+		List<Register> regs = new ArrayList<>();
+		if(e1 instanceof Register) {
+			regs.add((Register) e1);
+		}
+		if(e2 instanceof Register) {
+			regs.add((Register) e2);
+		}
+		return regs;
```

Comment:
Isn't it possible that `e1` and `e2` are complex expressions? Don't you need to traverse them for the registers via "e.getRegs"?.

Reply:
No, `e1` and `e2` are `LastValueInterface` which is only extended by `IConst`, `Location`, `MemoryObject` and `Register`.